src/components/mainScreen/MainScreen.tsx

```diff
@@ -1,7 +1,155 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+import queryString from 'query-string';
+
+import {
+  Card,
+  CardBody,
+  ClipboardCopy,
+  Slider,
+  Split,
+  SplitItem,
+  Stack,
+  StackItem,
+  Text,
+  TextVariants,
+  Title,
+  TitleSizes,
+} from '@patternfly/react-core';
 
 const MainScreen: React.FC = () => {
-  return <h1>AGUA</h1>;
+  const url = 'http://localhost:8080/';
+
+  const location = useLocation();
+
+  const { d, f } = queryString.parse(location.search);
+
+  const [frequencyOfTask, setFrequencyOfTask] = useState(f ? Number(f) : 0);
+  const [durationOfTask, setDurationOfTask] = useState(d ? Number(d) : 0);
+
+  const freqSteps = [
+    { value: 0, label: '50/day' },
+    { value: 1, label: '5/day' },
+    { value: 2, label: 'daily' },
+    { value: 3, label: 'weekly' },
+    { value: 4, label: 'monthly' },
+    { value: 5, label: 'yearly' },
+  ];
+
+  const durSteps = [
+    { value: 0, label: '1 second' },
+    { value: 1, label: '5 seconds' },
+    { value: 2, label: '30 seconds' },
+    { value: 3, label: '1 minute' },
+    { value: 4, label: '5 minutes' },
+    { value: 5, label: '30 minutes' },
+    { value: 6, label: '1 hour' },
+    { value: 7, label: '8 hours' },
+    { value: 8, label: '1 day' },
+  ];
+
+  const res = [
+    ['1 day', '2 hours', '30 minutes', '4 minutes', '1 minute', '5 seconds'],
+    ['5 days', '12 hours', '2 hours', '21 minutes', '5 minutes', '25 seconds'],
+    ['4 weeks', '3 days', '12 hours', '2 hours', '30 minutes', '2 minutes'],
+    ['8 weeks', '6 days', '1 day', '4 hours', '1 hour', '5 minutes'],
+    ['9 months', '4 weeks', '6 days', '21 hours', '5 hours', '25 minutes'],
+    ['NaN', '6 months', '5 weeks', '5 days', '1 day', '2 hours'],
+    ['NaN', '10 months', '2 months', '10 days', '2 days', '5 hour'],
+    ['NaN', 'NaN', 'NaN', '2 months', '2 weeks', '1 day'],
+    ['NaN', 'NaN', 'NaN', 'NaN', '8 weeks', '5 days'],
```

Comment:
Why use NaN here? I think we could have a better value for those cases.
The fixed matrix is good for the first version, but in the future we might do it dynamically

Reply:
Yes, this is just first implementation, just converting the table from the comics into the code. In future we can dynamically calculate times.